clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -760,18 +760,43 @@ public FetchManagerMetrics(Metrics metrics, String metricGrpPrefix) {
         }
 
         public void recordTopicFetchMetrics(String topic, int bytes, int records) {
+            Map<String, String> metricTags = new HashMap<>();
+            metricTags.put("topic", "topic-" + topic.replace(".", "_"));
```

Comment:
I feel we do not need the "topic-" prefix for the tag value as it will be shown as "tag-key"-"tag-value" already.